### PR TITLE
Allow fine-tuning the virtual floor style

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
The virtual floor by @JeroenDStout, introduced in the v0.1.2 release, has generally been well-received. However, some players find it too intrusive in its current appearance (cf. #7221). This PR gives them the option of a toned-down version, rather than having to turn it off completely.

screenshot_20180528_221350

The clear/transparent style, introduced by this PR, looks like the virtual floor as originally introduced in #6338:

The glassy/translucent style is the virtual floor as we've come to know it since the v0.1.2 release, and remains the default setting.

### CODE REVIEW

**Comment by janisozaur** [comment]
Have you thought of maybe taking #7575 into account in some way?

**Comment by AaronVanGeffen** [comment]
> What about drawing the grid on the surface for giving more of a sense what tile you're above?

Hmm… In a different, darker colour, perhaps? (Is there any demand for this?)

> And is it easy to adjust the thickness of the purple lines?

They're sprites, so we'd need additional sprites to offer different sizes.

**Comment by Broxzier** [comment]
> Hmm… In a different, darker colour, perhaps? (Is there any demand for this?)

In #7575 there's a vertical line that shows what tile the element is on:
image
RCT2 does have a similar thing with the white selection on the surface, but with the virtual floor enabled the purple grid can hide it completely, taking that sense away. Drawing the grid on the surface too may help get that sense back. I don't know if there is any demand for this, it was just something that I thought could improve the user experience even more.

> They're sprites, so we'd need additional sprites to offer different sizes.

I was afraid so. Maybe it's something for later, let's first hear if the players find this improvement enough.

File: src/openrct2-ui/input/Input.cpp [hunk 1]
```diff
@@ -176,7 +176,7 @@ void input_handle_keyboard(bool isTitle)
         }
     }
 
-    if (gConfigGeneral.use_virtual_floor)
+    if (gConfigGeneral.virtual_floor_style > 0)
```
Comment (Broxzier) [inline src/openrct2-ui/input/Input.cpp]: I'd rather have a check like `virtual_floor_style  != VIRTUAL_FLOOR_STYLE_DISABLED` or something similar.

File: src/openrct2/paint/VirtualFloor.cpp [hunk 2]
```diff
@@ -396,6 +397,9 @@ void virtual_floor_paint(paint_session * session)
             0, 0, 0, 1, _virtualFloorHeight, 5, 5, _virtualFloorHeight + ((dullEdges & 0x8) ? -2 : 0));
     }
 
+    if (gConfigGeneral.virtual_floor_style < 2)
```
Comment (Broxzier) [inline src/openrct2/paint/VirtualFloor.cpp]: And then check here for `virtual_floor_style  == VIRTUAL_FLOOR_STYLE_DISABLED`.

**Comment by AaronVanGeffen** [comment]
@Broxzier Thanks for the feedback. I've added an enum for the config values. Much better to look at, and verify, indeed!

**Pull request merged**